<launch>
  <arg name="front_left_6mm_obstacle_topic_name" default="/apollo/sensor/camera/obstacle/front_left_6mm" />
  <arg name="front_right_6mm_obstacle_topic_name" default="/apollo/sensor/camera/obstacle/front_right_6mm" />
  <arg name="front_left_25mm_obstacle_topic_name" default="/apollo/sensor/camera/obstacle/front_left_25mm" />
  <arg name="front_right_25mm_obstacle_topic_name" default="/apollo/sensor/camera/obstacle/front_right_25mm" />
  <arg name="front_12mm_traffic_topic_name" default="/apollo/sensor/camera/traffic/front_12mm" />
  <arg name="frame_rate" default="10"/>
  <arg name="trigger_internal" default="0" />
  <arg name="trigger_fps" default="10" />
  <arg name="flag_load_image" default="0" />

  <group ns="camera_front_left_6mm">
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" respawn="true" />

    <node pkg="nodelet" type="nodelet" name="sensor_camera_front_left_6mm" output="screen"
          args="load usb_cam/UsbCamNodelet camera_nodelet_manager" respawn="true" >
      <param name="video_device" value="/dev/video0" />
    	<param name="image_width" value="1920" />
    	<param name="image_height" value="1080" />
    	<param name="pixel_format" value="yuyv" />
      <param name="v4l_bind_id" type="string" value="1" />
      <param name="frame_rate" value="$(arg frame_rate)" />
    	<param name="camera_frame_id" value="usb_cam" />
    	<param name="io_method" value="mmap"/>
      <param name="camera_timeout" value="150" />
      <param name="error_code" value="11" />
      <param name="flag_load_image" value="$(arg flag_load_image)" />
      <param name="file_path_url" type="string" value="/apollo/data/img/6L" />
      <param name="camera_info_url" type="string" value="file://$(find usb_cam)/params/onsemi_front_left_6mm_intrinsics.yaml" />
      <param name="trigger_internal" value="$(arg trigger_internal)"/>
      <param name="trigger_fps" value="$(arg trigger_fps)" />
      <remap from="/camera_front_left_6mm/image_raw0" to="$(arg front_left_6mm_obstacle_topic_name)" />
      <remap from="/apollo/sensor/camera/obstacle/camera_info" to="$(arg front_left_6mm_obstacle_topic_name)/camera_info" />
    </node>
  </group>

  <group ns="camera_front_right_6mm">
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" respawn="true" />

    <node pkg="nodelet" type="nodelet" name="sensor_camera_front_right_6mm" output="screen"
          args="load usb_cam/UsbCamNodelet camera_nodelet_manager" respawn="true" >
      <param name="video_device" value="/dev/video1" />
    	<param name="image_width" value="1920" />
    	<param name="image_height" value="1080" />
    	<param name="pixel_format" value="yuyv" />
      <param name="v4l_bind_id" type="string" value="2" />
      <param name="frame_rate" value="$(arg frame_rate)" />
    	<param name="camera_frame_id" value="usb_cam" />
    	<param name="io_method" value="mmap"/>
      <param name="camera_timeout" value="150" />
      <param name="error_code" value="12" />
      <param name="flag_load_image" value="$(arg flag_load_image)" />
      <param name="file_path_url" type="string" value="/apollo/data/img/6R" />
	    <param name="camera_info_url" type="string" value="file://$(find usb_cam)/params/onsemi_front_right_6mm_intrinsics.yaml" />
      <param name="trigger_internal" value="$(arg trigger_internal)"/>
      <param name="trigger_fps" value="$(arg trigger_fps)" />
      <remap from="/camera_front_right_6mm/image_raw0" to="$(arg front_right_6mm_obstacle_topic_name)" />
      <remap from="/apollo/sensor/camera/obstacle/camera_info" to="$(arg front_right_6mm_obstacle_topic_name)/camera_info" />
    </node>
  </group>

  <group ns="camera_front_left_25mm">
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" respawn="true" />

    <node pkg="nodelet" type="nodelet" name="sensor_camera_front_left_25mm" output="screen"
          args="load usb_cam/UsbCamNodelet camera_nodelet_manager" respawn="true" >
      <param name="video_device" value="/dev/video2" />
    	<param name="image_width" value="1920" />
    	<param name="image_height" value="1080" />
    	<param name="pixel_format" value="yuyv" />
      <param name="v4l_bind_id" type="string" value="3" />
      <param name="frame_rate" value="$(arg frame_rate)" />
    	<param name="camera_frame_id" value="usb_cam" />
    	<param name="io_method" value="mmap"/>
      <param name="camera_timeout" value="150" />
      <param name="error_code" value="13" />
      <param name="flag_load_image" value="$(arg flag_load_image)" />
      <param name="file_path_url" type="string" value="/apollo/data/img/25L" />
      <param name="camera_info_url" type="string" value="file://$(find usb_cam)/params/onsemi_front_left_25mm_intrinsics.yaml" />
      <param name="trigger_internal" value="$(arg trigger_internal)"/>
      <param name="trigger_fps" value="$(arg trigger_fps)" />
      <remap from="/camera_front_left_25mm/image_raw0" to="$(arg front_left_25mm_obstacle_topic_name)" />
      <remap from="/apollo/sensor/camera/obstacle/camera_info" to="$(arg front_left_25mm_obstacle_topic_name)/camera_info" />
    </node>
  </group>

  <group ns="camera_front_right_25mm">
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" respawn="true" />

    <node pkg="nodelet" type="nodelet" name="sensor_camera_front_right_25mm" output="screen"
          args="load usb_cam/UsbCamNodelet camera_nodelet_manager" respawn="true" >
      <param name="video_device" value="/dev/video3" />
    	<param name="image_width" value="1920" />
    	<param name="image_height" value="1080" />
    	<param name="pixel_format" value="yuyv" />
      <param name="v4l_bind_id" type="string" value="4" />
      <param name="frame_rate" value="$(arg frame_rate)" />
    	<param name="camera_frame_id" value="usb_cam" />
    	<param name="io_method" value="mmap"/>
      <param name="camera_timeout" value="150" />
      <param name="error_code" value="14" />
      <param name="flag_load_image" value="$(arg flag_load_image)" />
      <param name="file_path_url" type="string" value="/apollo/data/img/25R" />
	    <param name="camera_info_url" type="string" value="file://$(find usb_cam)/params/onsemi_front_right_25mm_intrinsics.yaml" />
      <param name="trigger_internal" value="$(arg trigger_internal)"/>
      <param name="trigger_fps" value="$(arg trigger_fps)" />
      <remap from="/camera_front_right_25mm/image_raw0" to="$(arg front_right_25mm_obstacle_topic_name)" />
      <remap from="/apollo/sensor/camera/obstacle/camera_info" to="$(arg front_right_25mm_obstacle_topic_name)/camera_info" />
    </node>
  </group>

  <group ns="camera_front_12mm">
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" respawn="true" />

    <node pkg="nodelet" type="nodelet" name="sensor_camera_front_12mm" output="screen"
          args="load usb_cam/UsbCamNodelet camera_nodelet_manager" respawn="true" >
      <param name="video_device" value="/dev/video4" />
    	<param name="image_width" value="1920" />
    	<param name="image_height" value="1080" />
    	<param name="pixel_format" value="yuyv" />
      <param name="v4l_bind_id" type="string" value="2" />
      <param name="frame_rate" value="$(arg frame_rate)" />
    	<param name="camera_frame_id" value="usb_cam" />
    	<param name="io_method" value="mmap"/>
      <param name="camera_timeout" value="150" />
      <param name="error_code" value="15" />
      <param name="flag_load_image" value="$(arg flag_load_image)" />
      <param name="file_path_url" type="string" value="/apollo/data/img/6R" />
      <param name="camera_info_url" type="string" value="file://$(find usb_cam)/params/onsemi_front_12mm_intrinsics.yaml" />
      <param name="trigger_internal" value="0"/>
      <param name="trigger_fps" value="$(arg trigger_fps)" />
      <remap from="/camera_front_12mm/image_raw0" to="$(arg front_12mm_traffic_topic_name)" />
      <remap from="/apollo/sensor/camera/traffic/camera_info" to="$(arg front_12mm_traffic_topic_name)/camera_info" />
    </node>
  </group>

</launch>
